pkg/migrate/azure/privatedns.go

```diff
@@ -0,0 +1,744 @@
+				Fqdn:        legacyRecordSet.Fqdn,
+				SoaRecord:   legacySoaRecordToPrivate(legacyRecordSet.SoaRecord),
+				CnameRecord: legacyCnameRecordToPrivate(legacyRecordSet.CnameRecord),
+				MxRecords:   legacyMxRecordsToPrivate(legacyRecordSet.MxRecords),
+				ARecords:    legacyARecordsToPrivate(legacyRecordSet.ARecords),
+				AaaaRecords: legacyAaaaRecordsToPrivate(legacyRecordSet.AaaaRecords),
+				PtrRecords:  legacyPtrRecordsToPrivate(legacyRecordSet.PtrRecords),
+				SrvRecords:  legacySrvRecordsToPrivate(legacyRecordSet.SrvRecords),
+				TxtRecords:  legacyTxtRecordsToPrivate(legacyRecordSet.TxtRecords),
+			},
+			Type: &recordType,
+		}
+		privateRecordSets = append(privateRecordSets, &privateRecordSet)
+	}
+
+	ctx, cancel := context.WithTimeout(context.TODO(), 300*time.Second)
+	defer cancel()
+
+	// Create/Update the Zone
+	logrus.Printf("zone: %s ... ", *privateZone.Name)
+	zoneFuture, err := client.zonesClient.CreateOrUpdate(ctx, client.resourceGroup, *privateZone.Name, privateZone, "", "")
+	if err != nil {
+		return err
+	}
+
+	// Wait for zone creation to complete
+	err = zoneFuture.WaitForCompletionRef(ctx, client.zonesClient.Client)
+	if err != nil {
+		return err
+	}
+
+	// Read back the newly created zone to verify creation
+	_, err = client.zonesClient.Get(ctx, client.resourceGroup, *privateZone.Name)
+	if err != nil {
+		return err
+	}
+	logrus.Printf("ok.\n")
+
+	for _, recordSet := range privateRecordSets {
+		recordType := azprivatedns.RecordType(strings.TrimPrefix(*recordSet.Type, "Microsoft.Network/privateDnsZones/"))
+		relativeRecordSetName := *recordSet.Name
+		recordSet.Type = nil
+
+		// Create/Update the record
+		logrus.Printf("record: %s %s ... ", recordType, relativeRecordSetName)
+		_, err := client.recordsetsClient.CreateOrUpdate(ctx, client.resourceGroup, *privateZone.Name, recordType, relativeRecordSetName, *recordSet, "", "")
+		if err != nil {
+			return err
+		}
+
+		// Read back the newly created record to verify creation
+		_, err = client.recordsetsClient.Get(ctx, client.resourceGroup, *privateZone.Name, recordType, relativeRecordSetName)
+		if err != nil {
+			return err
+		}
+		logrus.Printf("ok.\n")
+	}
+
+	// Do we link, or not?
+	if link == false || client.virtualNetwork == "" {
+		return nil
+	}
+
+	// Get the virtual network so we have some parameters for the link creation
+	virtualNetwork, err := client.virtualNetworksClient.Get(ctx, client.vnetResourceGroup, client.virtualNetwork, "")
+	if err != nil {
+		return err
+	}
+
+	location := "global"
+	virtualNetworkID := *virtualNetwork.ID
+	virtualNetworkLinkName := fmt.Sprintf("%s-network-link", strings.Replace(client.vnetResourceGroup, "-rg", "", 1))
+
+	// pass a flag for this?
+	registrationEnabled := false
+
+	virtualNetworkLink := azprivatedns.VirtualNetworkLink{
+		Location: &location,
+		VirtualNetworkLinkProperties: &azprivatedns.VirtualNetworkLinkProperties{
+			VirtualNetwork: &azprivatedns.SubResource{
+				ID: &virtualNetworkID,
```

Comment:
same as https://github.com/openshift/installer/pull/3343/files#r399574726

Reply:
Updated to use autorest/to.